### PR TITLE
add support for pki auth

### PR DESCRIPTION
**Pull request by realmgic**

### CODE REVIEW

File: observer.go [hunk 1]
```diff
@@ -101,8 +101,11 @@ func newObserver(server *aero.Host, user, pass string) (o *Observer, err error)
 	clientPolicy := aero.NewClientPolicy()
 	clientPolicy.User = string(username)
 	clientPolicy.Password = string(password)
+
 	if authMode == "external" {
 		clientPolicy.AuthMode = aero.AuthModeExternal
+	} else if authMode == "pki" {
+		clientPolicy.AuthMode = aero.AuthModePKI
```
Comment (spkesan) [inline observer.go:108]:
```suggestion
        switch authMode {
        case "internal":
		clientPolicy.AuthMode = aero.AuthModeInternal
        case "external":
		clientPolicy.AuthMode = aero.AuthModeExternal
	case "pki":
		clientPolicy.AuthMode = aero.AuthModePKI
        default:
		log.Fatalln("Invalid auth mode: only `internal`, `external`, `pki` values are accepted.")
    }
```
and remove line 85-86

**Review by spkesan — COMMENTED** [review]
What do you think about having validations in conjunction with TLS configurations when PKI auth mode is used. I'll let @mphanias chip in.

File: observer.go (hunk 1 above)
Comment (realmgic) [reply observer.go]: Fixed.  I also added the default value of "" as an option.

**Comment by spkesan** [comment]
Thanks @realmgic .

Two things,
1. Can you create this PR against `dev` branch ?
2. And, we might want to add a validation such that when `authMode` is `pki`, `clientPolicy.TlsConfig.Certificates` (which contains the certiticates that the client presents to the server) must not be empty. What do you think?

File: observer.go [hunk 2]
```diff
@@ -101,8 +95,21 @@ func newObserver(server *aero.Host, user, pass string) (o *Observer, err error)
...
+
+	authMode := strings.ToLower(strings.TrimSpace(config.Aerospike.AuthMode))
+
+	switch authMode {
+	case "internal", "":
+		clientPolicy.AuthMode = aero.AuthModeInternal
+	case "external":
 		clientPolicy.AuthMode = aero.AuthModeExternal
+	case "pki":
+		if len(config.Aerospike.RootCA) == 0 || len(config.Aerospike.CertFile) == 0 || len(config.Aerospike.KeyFile) == 0 {
```
[the remark this answers is not in the document]
Comment (realmgic) [reply observer.go:107]: I was following this pattern from the `initAerospikeTLS`:
https://github.com/aerospike/aerospike-prometheus-exporter/blob/e137a689813149882a4d29d8f94fa431bdf244cd/observer.go#L43-L45

Is it not correct there?

**Pull request merged**